tests/python/unittest/test_gluon.py

```diff
@@ -530,13 +562,19 @@ def __init__(self, **kwargs):
         model.collect_params()
         assert len(w) == 0
 
+
+@with_seed()
 def test_sequential_warning():
     with warnings.catch_warnings(record=True) as w:
+        # The following line permits the test to pass if run multiple times
+        warnings.simplefilter('always')
```

Comment:
In what case would a test run multiple times and why shall it fail in that case?

Reply:
This PR provides an easy way to run a test multiple times under different seeds.  I found that this particular test would fail under those circumstances without this fix, when invoked as:

MXNET_TEST_COUNT=100 nosetests --verbose tests/python/unittest/test_gluon.py

Reply:
I think this sounds like a valid failure that should be resolved. Imagine a user calling the same method in a server and failing after some time because we might have some memory leak or something else that causes this error.

Reply:
The default behavior of 'warnings' is to collapse identical messages.  This test purposefully does something that it thinks should generate a message.  For the 'MXNET_TEST_COUNT=100' example above, it would not.  Thus, the test should ensure, independent of the global policy, that identical messages are not squashed.  Try the following:
```
python
Python 2.7.12 (default, Nov 20 2017, 18:23:56) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import mxnet as mx
>>> import warnings
>>> from mxnet import gluon
>>> warnings.simplefilter('always')
>>> b = gluon.nn.Sequential()
>>> b.add(gluon.nn.Dense(20))
>>> b.hybridize()
/home/dcarter/mxnet_dev/dgx/mxnet/python/mxnet/gluon/nn/basic_layers.py:83: UserWarning: All children of this Sequential layer are HybridBlocks. Consider using HybridSequential for the best performance.
  warnings.warn('All children of this Sequential layer are HybridBlocks. Consider ' \
>>> b.hybridize()
/home/dcarter/mxnet_dev/dgx/mxnet/python/mxnet/gluon/nn/basic_layers.py:83: UserWarning: All children of this Sequential layer are HybridBlocks. Consider using HybridSequential for the best performance.
  warnings.warn('All children of this Sequential layer are HybridBlocks. Consider ' \
>>> 
```
If you start over and repeat this sequence, but leave out the 'warnings.simplefilter('always')' line, then the final 'b.hybridize()' line will output nothing.
I invite you to pursue making 'warnings.simplefilter('always')' part of all gluon usage, but independent of that, I think this test should leave this line in, since that's what the test needs to pass.

Reply:
I understand, thanks for the elaboration.